ergo-lib/src/wallet/signing.rs

```diff
@@ -260,6 +265,21 @@ pub fn sign_reduced_transaction(
     )?)
 }
 
+/// Sign arbitrary message under a key representing a statement provable via a sigma-protocol.
+/// A statement can be a simple ProveDlog (PK) or a complex sigma conjectives tree
+pub fn sign_message(
+    prover: &dyn Prover,
+    sigma_tree: SigmaBoolean,
+    msg: &[u8],
+) -> Result<Vec<u8>, ProverError> {
+    let proof = prover.generate_proof(sigma_tree, msg, &HintsBag::empty());
+
+    match proof {
```

Comment:
I believe the whole `match` expr can be swapped with a single `.map` on `proof`.